test/api-digester/Inputs/stdlib-stable-abi.json

```diff
@@ -177708,6 +177708,32 @@
       ],
       "funcSelfKind": "NonMutating"
     },
+    {
+      "kind": "Function",
+      "name": "_isspace_clocale",
+      "printedName": "_isspace_clocale(_:)",
+      "children": [
+        {
+          "kind": "TypeNominal",
+          "name": "Bool",
+          "printedName": "Bool",
+          "usr": "s:Sb"
+        },
+        {
+          "kind": "TypeNominal",
+          "name": "UInt8",
+          "printedName": "UInt8",
+          "usr": "s:s5UInt8V"
+        }
+      ],
+      "declKind": "Func",
+      "usr": "s:s16_isspace_clocaleySbs5UInt8VF",
+      "moduleName": "Swift",
+      "declAttributes": [
+        "Inlinable"
+      ],
+      "funcSelfKind": "NonMutating"
+    },
```

Comment:
This should be flagged as an ABI-breaking change. @benlangmuir and @nkcsgexi  would this be caught by the checker? This is a dangerous and very easy ABI break to make.

Reply:
The ABI checker doesn't flag additions at all. Should it?

Reply:
This is an non-versioned addition. Wouldn't we want to catch those?

Reply:
If we only cared about forward compatibility, that'd be fine, but sure, I guess we could say any new ABI-public thing needs availability.

Reply:
@milseman @jrose-apple could one of you provide a concise but comprehensive description of what the ABI checker should check for and flag as breaking ?

Reply:
Preferably with a radar as well.

Reply:
> If we only cared about forward compatibility, that'd be fine, but sure, I guess we could say any new ABI-public thing needs availability.

Sorry, I failed to parse out your sentiment. Do you agree that adding a new, non-versioned symbol breaks backwards deployment and thus should be caught by the ABI checker?

Reply:
- Adding a new symbol without availability breaks backwards-deployment.
- Adding a new symbol without availability does not break forwards-deployment.
- The ABI checker should probably check backwards-deployment as well as forwards-deployment by default.

Reply:
@akyrtzi rdar://problem/50262430

Any new symbol added, whether public, @usableFromInline, or @inlinable, needs to be versioned (i.e. have an @availability annotation). Specifically, it needs to be versioned with a number greater than the current shipping OS, for all OSes. The ABI checker needs to detect this otherwise additions can break backwards deployment.

Note that @_alwaysEmitIntoClient does not emit a new symbol (or at least is designed not to), so it does not need @availability.